containers/src/Kokkos_DualView.hpp

```diff
@@ -895,22 +895,18 @@ class DualView : public ViewTraits<DataType, Arg1Type, Arg2Type, Arg3Type> {
   /// This discards any existing contents of the objects, and resets
   /// their modified flags.  It does <i>not</i> copy the old contents
   /// of either View into the new View objects.
-  void realloc(const size_t n0 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
-               const size_t n1 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
-               const size_t n2 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
-               const size_t n3 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
-               const size_t n4 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
-               const size_t n5 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
-               const size_t n6 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
-               const size_t n7 = KOKKOS_IMPL_CTOR_DEFAULT_ARG) {
+  template <class... I>
+  void impl_realloc(const size_t n0, const size_t n1, const size_t n2,
+                    const size_t n3, const size_t n4, const size_t n5,
+                    const size_t n6, const size_t n7, const I&... arg_prop) {
     const size_t new_extents[8] = {n0, n1, n2, n3, n4, n5, n6, n7};
     const bool sizeMismatch =
         Impl::size_mismatch(h_view, h_view.rank_dynamic, new_extents);
 
     if (sizeMismatch) {
-      ::Kokkos::realloc(d_view, n0, n1, n2, n3, n4, n5, n6, n7);
+      ::Kokkos::realloc(arg_prop..., d_view, n0, n1, n2, n3, n4, n5, n6, n7);
       h_view = create_mirror_view(d_view);
-    } else
+    } else  // FIXME could avoid this for WithoutInitializing
```

Comment:
It's worth discussing if this OK to keep for now. Using C++17 fold expressions this would be easy to improve. Otherwise, I would need to write some kind of `all_values_true` helper function.